recipes/bacnet-stack/all/conanfile.py

```diff
@@ -0,0 +1,64 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class BacnetStackConan(ConanFile):
+    name = "bacnet-stack"
+    license = "BSD"
+    url = "https://github.com/bacnet-stack/bacnet-stack/"
+    description = """
+        BACnet Protocol Stack library provides a BACnet application layer,
+        network layer and media access (MAC) layer communications services."""
+    topics = ("bacnet")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
```

Comment:
Cache the `cmake` object so reconfiguration is not needed in `build` and `package`.
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
```